contracts/profit/src/state/idle.rs

```diff
@@ -117,11 +117,19 @@ impl Enterable for Idle {
 }
 
 impl ConfigManagement for Idle {
-    fn try_update_config(self, cadence_hours: CadenceHours) -> ContractResult<Self> {
-        Ok(Self {
-            config: self.config.update(cadence_hours),
-            ..self
-        })
+    fn with_config<F>(self, f: F) -> ContractResult<StateAndResponse<Self>>
+    where
+        F: FnOnce(Config) -> ContractResult<StateAndResponse<Config>>,
+    {
+        f(self.config).map(
+            |StateAndResponse {
+                 state: config,
```

Comment:
obviously, passing the new config via the 'state' is a hack we do not want in our code.
Please do not misuse State Machine State for other purposes.

Reply:
I used the same structure to reuse code, but your concern is valid. I'll fork it into a separate one for the config.